pages/index.tsx

```diff
@@ -1,25 +1,29 @@
-import type { NextPage } from "next";
+import type { GetServerSideProps, NextPage } from "next";
 import Head from "next/head";
 import Image from "next/image";
 import styles from "../styles/Home.module.css";
 import { AnchorCard } from "../modules/cards";
 
+type Data = {
+  name: string;
+};
+
 export async function getServerSideProps() {
-  const res = await fetch(`http://localhost:3000/api/hello`)
-  const data = await res.json()
+  const res = await fetch(`http://localhost:3000/api/hello`);
+  const data = await res.json();
 
   if (!data) {
     return {
       notFound: true,
-    }
+    };
   }
 
   return {
     props: data,
-  }
+  };
 }
 
-const Home: NextPage = (props) => {
+const Home: NextPage<Data> = (props) => {
   return (
     <div className={styles.container}>
```

Comment:
I have no idea of how to type this, and full disclosure... didn't read further than the type hint on `NextPage` :joy: